update.sh

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env bash
+
+set -e
+
+echo "_______________________UPDATING INFRASTRUCTURE_______________________"
+
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+check_local_changes() {
+  local dir=$1
+  local repo_path="$SCRIPT_DIR/$dir"
+  
+  if [ ! -d "$repo_path" ]; then
+    echo "Directory $repo_path does not exist. Skipping..."
+    return
+  fi
+
+  cd "$repo_path"
+
+  git update-index -q --refresh
+
+  if [ -n "$(git status --porcelain)" ]; then
+    echo "Local changes detected in $dir. Please commit, stash, or discard your changes before updating."
+    exit 1 
+  fi
+  cd "$SCRIPT_DIR"
```

Comment:
Just an FYI in case you want to give it another go, but bash provides `pushd` and `popd` utils, which basically remembers all the paths that you traversed during the script, which should make the `SCRIPT_DIR` variable unnecessary.

Not sure if they exist in Mac though

Reply:
The installation scripts will be updated soon due to new proposed changes, this will be taken into account later, thank you!

Reply:
P.S. It does exist on Mac